src/categories/categories.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { log } from 'console';
+import { AppModule } from 'src/app.module';
+import { Repository } from 'typeorm';
+import { Category } from './Category';
+
+@Injectable()
+export class CategoriesService {
+    constructor(@InjectRepository(Category) private categoryRepository: Repository<Category>){}  
+    categories:Category[]
+
+    async createCategory(item:Category){
+        const {name , icon , color} = item
+        const category = await this.categoryRepository.save({name , icon , color})
+        return category
+    }
+
+    async updateCategory(id:number ,item:Category){
+        const {name , icon , color} = item
+        const getCategory = await this.categoryRepository.findOneBy({id})
+
+        if (getCategory) {
+            await this.categoryRepository.save({name , icon , color ,id:getCategory.id})
+            return {message: "updated"}
+        }
+        else {
+            return {message: "id yok"}
```

Comment:
This is correct, but remind me in the next session to talk about hacking in that regard.